pkg/restapi/rp/operation/operations.go

```diff
@@ -968,52 +970,44 @@ func (o *Operation) handlePresentationMsg(msg service.DIDCommMsg) error {
 
 	logger.Debugf("handling present-proof message: %+v", presentation)
 
-	vp, err := parsePresentationSubmission(presentationSubmissionFormat, presentation)
+	presentationSubmissionVP, err := getPresentationSubmissionVP(presentationSubmissionFormat, presentation)
 	if err != nil {
 		return fmt.Errorf("failed to parse verifiable presentation : %w", err)
 	}
 
-	// TODO parse and validate presentation_submission returned by the issuer
-	logger.Debugf("received presentation_submission : %+v", vp)
+	logger.Debugf("received presentation_submission : %+v", presentationSubmissionVP)
+
+	err = evaluatePresentationSubmission(presentationSubmissionVP)
+	if err != nil {
+		return fmt.Errorf("failed to evaluate presentation submission : %w", err)
+	}
 
 	return nil
 }
 
-// TODO json-ld context for issuer response: https://github.com/trustbloc/edge-adapter/issues/111
-func parsePresentationSubmission(
-	attachmentFormatID string, presentation *presentproof.Presentation) (*verifiable.Presentation, error) {
-	var attachmentID string
-
-	for _, f := range presentation.Formats {
-		if f.Format == attachmentFormatID {
-			attachmentID = f.AttachID
-		}
-	}
-
-	if attachmentID == "" {
-		return nil, fmt.Errorf("no attachment found for given format %s", attachmentFormatID)
+func evaluatePresentationSubmission(p *rpvc.PresentationSubmissionPresentation) error {
+	rawCreds, err := p.Base.MarshalledCredentials()
+	if err != nil {
+		return fmt.Errorf("failed to marshal raw credentials from presentation : %w", err)
 	}
 
-	a := getAttachmentByID(attachmentID, presentation.PresentationsAttach)
-	if a == nil {
-		return nil, fmt.Errorf("attachment referenced by ID %s from a format was not found", attachmentID)
+	if len(rawCreds) != 1 {
+		return fmt.Errorf("expected 1 VC in presentation but got %d", len(rawCreds))
 	}
 
-	vpBytes, err := a.Data.Fetch()
+	vc, err := verifiable.ParseCredential(rawCreds[0])
 	if err != nil {
-		return nil, fmt.Errorf("failed to fetch contents of attachment with id %s : %w", attachmentID, err)
+		return fmt.Errorf("%w : failed to parse raw credential : %s", errMalformedCredential, err)
 	}
 
-	return verifiable.ParsePresentation(vpBytes)
-}
-
-func getAttachmentByID(id string, attachments []decorator.Attachment) *decorator.Attachment {
-	for i := range attachments {
-		if attachments[i].ID == id {
-			return &attachments[i]
-		}
+	if !adapterutil.StringsContains(rpvc.CreditCardStatementCredentialType, vc.Types) {
```

Comment:
Here, we still have domain specific `CreditCardStatement` hardcoded in main codebase.

Reply:
@rolsonquadras removed